### PR TITLE
feature(lxy): add popart & value rescale & symlog to ppof

### PR DESCRIPTION
**Pull request by karroyan**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/bonus/ppof.py [hunk 1]
```diff
@@ -70,10 +70,23 @@ def __init__(
             action_shape = get_hybrid_shape(action_space)
         else:
             action_shape = action_space.shape
+        assert self.cfg.value_norm in ['popart', 'value_rescale', 'symlog']
```
Comment (PaParaZz1) [inline ding/bonus/ppof.py:73]: add assertion info

File: ding/bonus/ppof.py [hunk 2]
```diff
@@ -70,10 +70,23 @@ def __init__(
...
+                    action_shape,
+                    action_space=self.cfg.action_space,
+                    **self.cfg.model
+                )
+            else:
+                model = PPOFModel(
+                    self.env.observation_space.shape,
+                    action_shape,
+                    action_space=self.cfg.action_space,
+                    use_popart_head=True,
```
Comment (PaParaZz1) [inline ding/bonus/ppof.py:87]: just name `popart_head`

File: ding/model/common/head.py [hunk 3]
```diff
@@ -1170,6 +1170,75 @@ def forward(self, x: torch.Tensor) -> Dict:
...
+            - x: :math:`(B, N)`, where ``B = batch_size`` and ``N = hidden_size``.
+            - logit: :math:`(B, M)`, where ``M = output_size``.
+        Examples:
+            >>> head = PopArtQHead(64, 64)
+            >>> inputs = torch.randn(4, 64)
+            >>> outputs = head(inputs)
+            >>> assert isinstance(outputs, dict) and outputs['logit'].shape == torch.Size([4, 64])
+        """
+        x = self.Q(x)
+        return x
```
Comment (PaParaZz1) [inline ding/model/common/head.py:1239]: return the same dict like regression head

File: ding/rl_utils/value_rescale.py [hunk 4]
```diff
@@ -18,3 +18,19 @@ def value_inv_transform(x: torch.Tensor, eps: float = 1e-2) -> torch.Tensor:
         :math: `h^{-1}(x) = sign(x)({(\frac{\sqrt{1+4\eps(|x|+1+\eps)}-1}{2\eps})}^2-1)` .
     """
     return torch.sign(x) * (((torch.sqrt(1 + 4 * eps * (torch.abs(x) + 1 + eps)) - 1) / (2 * eps)) ** 2 - 1)
+
```
Comment (PaParaZz1) [inline ding/rl_utils/value_rescale.py:21]: add unittest

File: ding/rl_utils/value_rescale.py [hunk 5]
```diff
@@ -18,3 +18,19 @@ def value_inv_transform(x: torch.Tensor, eps: float = 1e-2) -> torch.Tensor:
...
+
+def symlog(x: torch.Tensor) -> torch.Tensor:
+    r"""
+    Overview:
+        :math: `symlog(x) = sign(x)(\ln{|x|+1})`
```
Comment (PaParaZz1) [inline ding/rl_utils/value_rescale.py:26]: add paper link

File: ding/rl_utils/value_rescale.py [hunk 6]
```diff
@@ -18,3 +18,19 @@ def value_inv_transform(x: torch.Tensor, eps: float = 1e-2) -> torch.Tensor:
...
+    """
+    return torch.sign(x) * (torch.log(torch.abs(x) + 1))
+
+
+def inv_symlog(x: torch.Tensor) -> torch.Tensor:
+    r"""
```
Comment (PaParaZz1) [inline ding/rl_utils/value_rescale.py:32]: add more comments like our format

File: ding/torch_utils/network/popart.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+**POP**: to preserve the outputs of the unnormalized function when we change the scale and shift.
+
+"""
+from typing import Optional, Union
+import math
+import torch
+import torch.nn as nn
+
+
+class PopArt(nn.Module):
```
Comment (PaParaZz1) [inline ding/torch_utils/network/popart.py:17]: add unittest

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/model/common/head.py (hunk 3 above)
Comment (karroyan) [reply ding/model/common/head.py:1239]: The x that popart network returns is already a dict of keys {'pred'， ‘unnormalized_pred’} : https://github.com/karroyan/DI-engine/blob/cd5d31afabcdc2daa97a5308f079623c38d75c73/ding/torch_utils/network/popart.py#L64

File: ding/torch_utils/network/tests/test_popart.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@pytest.mark.unittest
+class TestPopArt:
+
+    def test_popart(self):
+        input = torch.rand((batch_size, input_features)).requires_grad_(True)
+        model = PopArt(input_features, output_features)
+        output = model(input)
+
```
Comment (PaParaZz1) [inline ding/torch_utils/network/tests/test_popart.py:17]: assert grad

File: ding/model/common/head.py (hunk 3 above)
Comment (PaParaZz1) [reply ding/model/common/head.py:1239]: add some comments about the structure of x

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**